frontend/app/src/components/home/AccountInfo.svelte

```diff
@@ -26,19 +34,96 @@
         fullWidthOnMobile = false,
     }: Props = $props();
 
+    const client = getContext<OpenChat>("client");
+
     let tokenDetails = $derived($cryptoLookup.get(ledger)!);
-    let account = $derived(
-        tokenDetails.symbol === ICP_SYMBOL ? $currentUserStore.cryptoAccount : $currentUserIdStore,
-    );
-    let symbol = $derived(tokenDetails.symbol);
+    let selectedNetwork = $state<string | undefined>(undefined);
+    let networks = $derived.by(() => {
+        if (tokenDetails.symbol === BTC_SYMBOL) {
+            return [BTC_SYMBOL, CKBTC_SYMBOL];
+        } else {
+            return [];
+        }
+    });
+    $effect(() => {
+        selectedNetwork = networks[0];
+    });
+
+    let account = $derived.by(() => {
+        if (tokenDetails.symbol === ICP_SYMBOL) {
+            return $currentUserStore.cryptoAccount;
+        } else if (tokenDetails.symbol === BTC_SYMBOL && selectedNetwork === BTC_SYMBOL) {
+            return $bitcoinAddress;
+        } else {
+            return $currentUserIdStore;
+        }
+    });
+
+    const btcAddressPromise = new Lazy(() => client.getBtcAddress());
+
+    let error = $state();
+    $effect(() => {
+        if (account === undefined) {
+            if (selectedNetwork === BTC_SYMBOL) {
+                btcAddressPromise.get().catch((e) => error = e);
```

Comment:
This seems a bit odd - getting the btc address and then doing ... nothing with it. Feels fishy.

Reply:
When the promise completes it'll set the `btcAddress` store which `account` is derived from (if the BTC network is still selected).

Reply:
On line 56

Reply:
It's not the source of truth so much as the control flow which is about as confusing as could be. We have a thing called `btcAddressPromise` which is _not_ a promise. Then we call `get` on it but we do nothing with the response. It's quite weird. Feels like we could just call `client.getBtcAddress` when required and put the result in a `$state` and the whole thing would be a lot less confusing.

Reply:
Yeah makes sense, have just updated it, see what you think of the new version